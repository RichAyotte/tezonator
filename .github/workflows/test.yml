name: Run Tests

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Bun
      run: |
        curl -fsSL https://bun.sh/install | bash
        echo "$HOME/.bun" >> $GITHUB_PATH

    - name: Check Bun version
      run: bun --version

    - name: Install dependencies
      run: bun install

    - name: Run tests
      run: bun test
