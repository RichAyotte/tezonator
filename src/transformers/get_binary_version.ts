import type { ParsedPath } from 'node:path'

type BinaryVersion = {
	bin_path: ParsedPath
	commit_hash?: string
	date?: Date
	version?: string
}

const regex =
	/(\w+) \((\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \+\d{4})\) \(([ \w.~+]+)\)/

type GetBinaryVersionInput = {
	bin_path: ParsedPath
	stdout: string
}

export function get_binary_version({
	bin_path,
	stdout,
}: GetBinaryVersionInput): BinaryVersion {
	const match = stdout.trim().match(regex)

	if (!match) {
		throw new Error(
			`${bin_path} version could not be determined. ${stdout.trim()}`,
		)
	}

	const [, commit_hash, date, version] = match
	return {
		bin_path,
		commit_hash,
		date: new Date(date),
		version,
	}
}
