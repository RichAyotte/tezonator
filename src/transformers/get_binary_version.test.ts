import { describe, expect, test } from 'bun:test'
import { get_binary_version } from './get_binary_version'

const bin_path = {
	root: '/',
	dir: '/home/rich/.local/bin/tezonator/0a81ce76',
	base: 'octez-client',
	ext: '',
	name: 'octez-client',
}

describe('get_binary_version', () => {
	test('(Octez 0.0+dev)', () => {
		expect(
			get_binary_version({
				bin_path,
				stdout: '0a81ce76 (2024-03-19 17:09:20 +0000) (Octez 0.0+dev)\n',
			}),
		).toMatchSnapshot()
	})
	test('(18.0~rc1+dev)', () => {
		expect(
			get_binary_version({
				bin_path,
				stdout: 'cae5b795 (2024-02-29 10:34:23 +0000) (18.0~rc1+dev)\n',
			}),
		).toMatchSnapshot()
	})
})
