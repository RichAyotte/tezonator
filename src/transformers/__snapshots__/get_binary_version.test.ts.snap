// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`get_binary_version (Octez 0.0+dev) 1`] = `
{
  "bin_path": {
    "base": "octez-client",
    "dir": "/home/rich/.local/bin/tezonator/0a81ce76",
    "ext": "",
    "name": "octez-client",
    "root": "/",
  },
  "commit_hash": "0a81ce76",
  "date": 2024-03-19T17:09:20.000Z,
  "version": "Octez 0.0+dev",
}
`;

exports[`get_binary_version (18.0~rc1+dev) 1`] = `
{
  "bin_path": {
    "base": "octez-client",
    "dir": "/home/rich/.local/bin/tezonator/0a81ce76",
    "ext": "",
    "name": "octez-client",
    "root": "/",
  },
  "commit_hash": "cae5b795",
  "date": 2024-02-29T10:34:23.000Z,
  "version": "18.0~rc1+dev",
}
`;

exports[`get_binary_version (Octez 20.0~beta2) 1`] = `
{
  "bin_path": {
    "base": "octez-client",
    "dir": "/home/rich/.local/bin/tezonator/0a81ce76",
    "ext": "",
    "name": "octez-client",
    "root": "/",
  },
  "commit_hash": "c8e95f06",
  "date": 2024-04-02T16:21:31.000Z,
  "version": "Octez 20.0~beta2",
}
`;
